Storage/SQL/SQLiteRemoteClientsAndTabs.swift

```diff
@@ -155,6 +155,29 @@ public class SQLiteRemoteClientsAndTabs: RemoteClientsAndTabs {
         return Deferred(value: Result(success: clients))
     }
 
+    public func getTabsForClientWithGUID(guid: GUID?) -> Deferred<Result<[RemoteTab]>> {
+        var tabsSQL = "SELECT * FROM \(TableTabs) where client_guid "
+        var clientArgs: Args?
+        if let clientGUID = guid {
+            tabsSQL += "= ?"
+            clientArgs = [guid]
+        } else {
+            tabsSQL += "IS NULL"
+        }
+
+        log.debug("Looking for tabs for client with guid: \(guid)")
+        let tabsResult = db.runQuery(tabsSQL, args: clientArgs, factory: tabs.factory!).value
```

Comment:
Careful of using `.value`.

```
return db.runQuery(…) >>== { deferResult($0.asArray) }
```

I think that'll still work for the empty array.